extensions/amp-form/0.1/test/test-amp-form.js

```diff
@@ -330,7 +331,9 @@ describes.repeated('', {
         emailInput.setAttribute('required', '');
         form.appendChild(emailInput);
         sandbox.spy(form, 'checkValidity');
-        sandbox.stub(ampForm.xhr_, 'fetch').returns(Promise.resolve());
+        const {promise, stub} = stubWithCalledPromise(
```

Comment:
Like the idea, have a suggestion for an alternative API:

```
var stub = sandbox.stub(ampForm.xhr_, 'fetch').returns(Promise.resolve());
.....
.....
whenCalled(stub).then()....
```

also optionally:
```
// when called for the third time.
whenCalled(stub, 3).then()....
```

on the implementation side of `whenCalled`, you can use `poll` method we have in `iframe.js` (probably can move somewhere else) to check for `callCount` (poll already returns a promise). This should work for both stub and spy since they both have `callCount`.

WDYT?